src/main.rs

```diff
@@ -138,9 +164,19 @@ pub fn main() {
 
     if let Some("clippy") = std::env::args().nth(1).as_ref().map(AsRef::as_ref) {
         // this arm is executed on the initial call to `cargo clippy`
+
+        match std::env::args().nth(2).as_ref().map(AsRef::as_ref) {
```

Comment:
shouldn't this be `.skip(2).any(...)`? It's mostly irrelevant where the `--help` is passed, as long as it's somewhere in the arguments